cognite/client/data_classes/data_modeling/instances.py

```diff
@@ -0,0 +1,486 @@
+                                set to 0, the upsert will behave as an insert, so it will fail the bulk if the
+                                item already exists. If skipOnVersionConflict is set on the ingestion request,
+                                then the item will be skipped instead of failing the ingestion request.
+        sources (list[NodeOrEdgeData]): List of source properties to write. The properties are from the node and/or
+                        container the container(s) making up this node.
+    """
+
+    def __init__(
+        self,
+        space: str,
+        external_id: str,
+        existing_version: int = None,
+        sources: list[NodeOrEdgeData] = None,
+        **_: dict,
+    ):
+        super().__init__(space, external_id, "node", existing_version, sources)
+
+    def as_id(self) -> NodeId:
+        return NodeId(space=self.space, external_id=self.external_id)
+
+
+class Node(Instance):
+    """A node. This is the read version of the node.
+
+    Args:
+        space (str): The workspace for the node.a unique identifier for the space.
+        external_id (str): Combined with the space is the unique identifier of the node.
+        version (str): DMS version.
+        last_updated_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+        created_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+        deleted_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+                            Timestamp when the instance was soft deleted. Note that deleted instances are filtered out of query results, but present in sync results
+        properties
+    """
+
+    def __init__(
+        self,
+        space: str,
+        external_id: str,
+        version: str,
+        last_updated_time: int,
+        created_time: int,
+        deleted_time: int = None,
+        properties: dict[str, dict[str, dict[str, PropertyValue]]] = None,
+        **_: dict,
+    ):
+        super().__init__(space, external_id, version, last_updated_time, created_time, "node", deleted_time, properties)
+
+    def as_apply(self, source: ViewId | ContainerId, existing_version: int) -> NodeApply:
+        return NodeApply(
+            space=self.space,
+            external_id=self.external_id,
+            existing_version=existing_version,
+            sources=[
+                NodeOrEdgeData(source=source, properties=space_properties[source.identifier])  # type: ignore[arg-type]
+                for space_properties in self.properties.values()
+            ]
+            if self.properties
+            else None,
+        )
+
+    def as_id(self) -> NodeId:
+        return NodeId(space=self.space, external_id=self.external_id)
+
+
+class NodeUpdate(InstanceUpdate):
```

Comment:
I suggest `NodeApplyResult` rather than `NodeUpdate`. (And NodeApplyResultList, EdgeApplyResult, EdgeApplyResultList aswell)